### PR TITLE
Refactor tests Jul 03

### PR DESCRIPTION
**Pull request by AliakseiHlebka**

### CODE REVIEW

File: src/test/java/page/GoogleLoginPage.java [hunk 1]
```diff
@@ -9,7 +9,7 @@
 public class GoogleLoginPage extends AbstractGoogleMailPage {
 
     private static final String GOOGLE_MAIL_URL = "https://mail.google.com";
-    private static final String HOMEPAGE_URL = "https://mail.google.com/mail/u/0/#inbox";
+    WebDriverWait wait = new WebDriverWait(driver, 5);
```
Comment (68889) [inline src/test/java/page/GoogleLoginPage.java]: Make a private one, please

File: src/test/java/page/GoogleLoginPage.java [hunk 2]
```diff
@@ -27,17 +27,14 @@ public GoogleLoginPage(WebDriver driver) {
...
+    public GoogleMailHomePage loginToGoogleMail(String login, String password) {
         driver.get(GOOGLE_MAIL_URL);
         loginOrEmailTextfield.sendKeys(login);
         confirmEmailOrPhoneButton.click();
-        new WebDriverWait(driver, 3).until(ExpectedConditions.visibilityOf(passwordTextField));
+        wait.until(ExpectedConditions.visibilityOf(passwordTextField));
         passwordTextField.sendKeys(password);
         confirmPasswordButton.click();
-        new WebDriverWait(driver, 10).until(ExpectedConditions.urlMatches(HOMEPAGE_URL));
+        wait.until(ExpectedConditions.urlMatches(new GoogleMailHomePage(driver).getHomepageUrl()));
```
Comment (68889) [inline src/test/java/page/GoogleLoginPage.java]: I think it is better to use some constants to define GoogleMailHomePage

File: src/test/java/page/GoogleMailEmailDetailsPage.java [hunk 3]
```diff
@@ -40,6 +46,10 @@ public WebElement getEmailAddresseeTextField() {
         return emailAddresseeTextField;
     }
 
+//    public WebElement getEmailSubjectTextField() {
```
Comment (68889) [inline src/test/java/page/GoogleMailEmailDetailsPage.java]: Delete comments, unless necessary

File: src/test/java/page/GoogleMailHomePage.java [hunk 4]
```diff
@@ -10,6 +10,7 @@
 
 public class GoogleMailHomePage extends AbstractGoogleMailPage {
 
+    private static final String HOMEPAGE_URL = "https://mail.google.com/mail/u/0/#inbox";
```
Comment (68889) [inline src/test/java/page/GoogleMailHomePage.java]: Why do you need to make it private?

File: src/test/java/test/GoogleMailDeleteIncomingEmailAndRestoreToInboxTest.java [hunk 5]
```diff
@@ -32,20 +33,16 @@ public void googleMailDeleteIncomingEmailAndRestoreToInboxTest() {
...
-
-        new GoogleLoginPage(driver).loginToGoogleMail(login, password);
-
-        homePage.openInboxEmail(emailSender)
-                .deleteEmail();
+        GoogleMailHomePage homePage = new GoogleLoginPage(driver).loginToGoogleMail(login, password);
+        GoogleMailEmailDetailsPage emailDetailsPage = homePage.openInboxEmail(emailSender);
+        homePage = emailDetailsPage.deleteEmail();
 
         Assert.assertTrue(emailDetailsPage.getActionConfirmationPopup()
```
Comment (68889) [inline src/test/java/test/GoogleMailDeleteIncomingEmailAndRestoreToInboxTest.java]: use asserEquals here

File: src/test/java/test/GoogleMailDeleteIncomingEmailAndRestoreToInboxTest.java [hunk 6]
```diff
@@ -32,20 +33,16 @@ public void googleMailDeleteIncomingEmailAndRestoreToInboxTest() {
...
                 .getText().contains(deleteEmailTargetMessage), "Email was not moved to trash bin!");
 
-        homePage.openTrashBinPage()
-                .openDeletedEmail(emailSender)
-                .moveEmailFromTrashBinToInbox();
+        GoogleMailTrashBinPage trashBinPage = homePage.openTrashBinPage();
+        emailDetailsPage = trashBinPage.openDeletedEmail(emailSender);
+        homePage = emailDetailsPage.moveEmailFromTrashBinToInbox();
 
         Assert.assertTrue(emailDetailsPage.getActionConfirmationPopup()
```
Comment (68889) [inline src/test/java/test/GoogleMailDeleteIncomingEmailAndRestoreToInboxTest.java]: the same as above

File: src/test/java/test/GoogleMailSaveNewMailAsDraftAndDeleteTest.java [hunk 7]
```diff
@@ -32,30 +33,25 @@ public void googleMailSaveMailAsDraftTest() {
         String emailAddressee = "al.hlebka@gmail.com";
         String emailSubject = "Message from Webdriver";
         String emailBody = "Hello Aliaksei,\nthis message is generated by Webdriver.\nBest regards,\nAliaksei.";
-        GoogleLoginPage loginPage = new GoogleLoginPage(driver);
 
-        loginPage.loginToGoogleMail(login, password);
+        GoogleMailHomePage homePage = new GoogleLoginPage(driver).loginToGoogleMail(login, password);
 
-        Assert.assertSame(driver.getCurrentUrl(), loginPage.getHomepageUrl(), "Login failed");
+        Assert.assertSame(driver.getCurrentUrl(), homePage.getHomepageUrl(), "Login failed");
```
Comment (68889) [inline src/test/java/test/GoogleMailSaveNewMailAsDraftAndDeleteTest.java]: I do not think that it is an appropriate way to compare the Strings. Use another assert

File: src/test/java/test/GoogleMailSaveNewMailAsDraftAndDeleteTest.java [hunk 8]
```diff
@@ -32,30 +33,25 @@ public void googleMailSaveMailAsDraftTest() {
...
-                .enterEmailBody(emailBody)
-                .closeNewEmail()
-                .openDraftsPage()
-                .openDraftEmail(emailSubject);
+                .enterEmailBody(emailBody);
+        homePage = createNewEmailPage.closeNewEmail();
+        GoogleMailDraftsPage draftsPage = homePage.openDraftsPage();
+        GoogleMailEmailDetailsPage emailDetailsPage = draftsPage.openDraftEmail(emailSubject);
 
         Assert.assertSame(emailDetailsPage.getEmailAddresseeTextField().
```
Comment (68889) [inline src/test/java/test/GoogleMailSaveNewMailAsDraftAndDeleteTest.java]: the same

File: src/test/java/page/GoogleLoginPage.java (hunk 1 above)
Comment (AliakseiHlebka) [reply src/test/java/page/GoogleLoginPage.java]: Made private

File: src/test/java/page/GoogleMailEmailDetailsPage.java (hunk 3 above)
Comment (AliakseiHlebka) [reply src/test/java/page/GoogleMailEmailDetailsPage.java]: Comment is removed

File: src/test/java/page/GoogleMailHomePage.java (hunk 4 above)
Comment (AliakseiHlebka) [reply src/test/java/page/GoogleMailHomePage.java]: Access modifier changed to package private

File: src/test/java/page/GoogleLoginPage.java (hunk 2 above)
Comment (AliakseiHlebka) [reply src/test/java/page/GoogleLoginPage.java]: Access modifier for constant from GoogleMailHomePage changed to package private to make it possible to access the constant here.

File: src/test/java/test/GoogleMailDeleteIncomingEmailAndRestoreToInboxTest.java (hunk 5 above)
Comment (AliakseiHlebka) [reply src/test/java/test/GoogleMailDeleteIncomingEmailAndRestoreToInboxTest.java]: Method changed to assertEquals()

File: src/test/java/test/GoogleMailDeleteIncomingEmailAndRestoreToInboxTest.java (hunk 6 above)
Comment (AliakseiHlebka) [reply src/test/java/test/GoogleMailDeleteIncomingEmailAndRestoreToInboxTest.java]: Method changed to assertEquals()

File: src/test/java/test/GoogleMailSaveNewMailAsDraftAndDeleteTest.java (hunk 7 above)
Comment (AliakseiHlebka) [reply src/test/java/test/GoogleMailSaveNewMailAsDraftAndDeleteTest.java]: Method changed to assertEquals()

File: src/test/java/test/GoogleMailSaveNewMailAsDraftAndDeleteTest.java (hunk 8 above)
Comment (AliakseiHlebka) [reply src/test/java/test/GoogleMailSaveNewMailAsDraftAndDeleteTest.java]: Method changed to assertEquals()

**Pull request merged**